src/NLog/LayoutRenderers/ExceptionLayoutRenderer.cs

```diff
@@ -176,8 +224,8 @@ protected override void Append(StringBuilder builder, LogEventInfo logEvent)
         /// Appends the Message of an Exception to the specified <see cref="StringBuilder" />.
         /// </summary>
         /// <param name="sb">The <see cref="StringBuilder"/> to append the rendered data to.</param>
-        /// <param name="ex">The exception containing the Message to append.</param>
-        protected virtual void AppendMessage(StringBuilder sb, Exception ex)
+        /// <param name="ex">The exception containing the Message to append.</param>        
+        private static void AppendMessage(StringBuilder sb, Exception ex)
```

Comment:
There are two options now

1. Remove the Dict (`_renderingfunctions`) which will fall back to the loop and the switch case.
2. Change all the methods back to virtual method and make the dictionary as INSTANCE type rather than a static variable. 

When I try to change the method to public virtual, the dictionary isn't allowing. Which is understandable. I see the below error

`A field initializer cannot reference the non-static field, method, or property 'NLog.LayoutRenderers.ExceptionLayoutRenderer.AppendMessage(System.Text.StringBuilder, System.Exception)'`